argo/src/Argo/DefaultMain.hs

```diff
@@ -284,14 +287,20 @@ maxOccupancyOpt =
    (Opt.long "max-occupancy" <>
     Opt.help "Maximum number of active sessions allowed at once." <>
```

Comment:
Based on other comments about this option in this repo and in your PR, shouldn't this be "Maximum number of states allowed at once"? That seems like an important difference.

Reply:
Technically from an internal view I would say you're 100% right. Practically though, from a user point of view, you never see the "states" (they're implementation details of the connection/interactions), so I'm biased towards a description more phrased like how it is now. As far as a user/connection is concerned, it _is_ the number of active sessions allowed I think it's fair to say.

I'm happy to rephrase it -- I can't think off the top of my head on a different wording that avoids too much low level detail but still speaks to the observable behavior for clients.

Ah, but even if we leave the documentation string as is... it's probably worth adding a comment near that in the Haskell code to clarify for a developer reading that! That is confusing without all that other history/context/etc.